runelite-client/src/main/java/net/runelite/client/plugins/runecraft/ZeahOverlay.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2018, John Pettenger
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.runecraft;
+
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+import javax.imageio.ImageIO;
+import javax.inject.Inject;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.Client;
+import net.runelite.api.GameObject;
+import net.runelite.api.Varbits;
+import net.runelite.api.coords.LocalPoint;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayUtil;
+
+@Slf4j
+public class ZeahOverlay extends Overlay
+{
+	private static final int MAX_DISTANCE = 2600;
+	private static final int Z_OFFSET = 200;
+
+	private static final BufferedImage PICKAXE_IMAGE;
+
+	private final Client client;
+	private final RunecraftPlugin plugin;
+	private final RunecraftConfig config;
+
+	static
+	{
+		synchronized (ImageIO.class)
+		{
+			BufferedImage pickaxeImage;
+			try
+			{
+				pickaxeImage = ImageIO.read(ZeahOverlay.class.getResourceAsStream("/skill_icons/mining.png"));
+			}
+			catch (IOException e)
+			{
+				log.warn("Error loading mining skill icon", e);
+				pickaxeImage = null;
+			}
+			PICKAXE_IMAGE = pickaxeImage;
+		}
+
+	}
+
+	@Inject
+	private ZeahOverlay(Client client, RunecraftPlugin plugin, RunecraftConfig config)
+	{
+		this.client = client;
+		this.plugin = plugin;
+		this.config = config;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		if (!config.showDenseRunestoneIndicator())
```

Comment:
Seeing as the two dense essence mining spots are within the same region, we should also gate on being within the region. (see tears of guthix plugin)